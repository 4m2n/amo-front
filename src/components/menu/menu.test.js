import { render, fireEvent } from "@testing-library/react"
import { pipe, prop, tap } from "ramda"
import Menu from "./menu"
import React from "react"

describe("components :: menu", () => {
  it("should be closed by default", () => pipe(
      render,
      prop("container"),
      container => container.querySelector("nav"),
      prop("className"),
      className => expect(className).toBe("closed"),
  )(<Menu />))

  it("should be opened once the user clicked on the burger menu", () => pipe(
    render,
    tap(render => fireEvent.click(render.container.querySelector(".menu-burger"))),
    prop("container"),
    container => container.querySelector("nav"),
    prop("className"),
    className => expect(className).toBe("opened"),
  )(<Menu />))
})
