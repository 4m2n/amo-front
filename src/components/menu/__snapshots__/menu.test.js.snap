// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components :: menu should have an active menu item when the page matches the current path 1`] = `
<li
  className="active"
>
  <a
    href="/concerts"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    concerts
  </a>
  <div
    className="separator"
  />
</li>
`;

exports[`components :: menu should have an inactive menu item when the page doesn't matches the current path 1`] = `
<li
  className=""
>
  <a
    href="/biographie"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    biographie
  </a>
  <div
    className="separator"
  />
</li>
`;
